optuna/study/study.py

```diff
@@ -445,7 +445,9 @@ def objective(trial):
             RuntimeError:
                 If nested invocation of this method occurs.
         """
-
+        if ((n_trials == 0) and (timeout is None)) & (show_progress_bar==True):
+            warnings.warn("n_trials and timeout is set to None.Progress bar won't be displayed")
+            
```

Comment:
By following https://github.com/optuna/optuna/issues/4679, `n_trials` should be `None`. In addition, the condition can be simplified as follows:
```suggestion
        if show_progress_bar and n_trials is None and timeout is None:
            warnings.warn("n_trials and timeout is set to None.Progress bar won't be displayed")
       
```

Reply:
Hey. One question, If n_trials and timeout is None, to test this, we would need a triggering condition to stop it. Otherwise, won't it run indefinitely?